### ISSUE TITLE
Bug Report: HTTP 500 Error, then no response from the plug-in, then it might be working?

### ISSUE THREAD

**Issue by rdeutsch3** [bloomkd46/homebridge-XfinityHome#44]
### Describe The Bug:

Happened to be randomly staring at the log and saw an http 500 error from the Xfinity Home plug-in.  So I tried arming the system after a few moments and there was nothing in the logs and the alarm panel on the wall did not change state.  So I tried arming it again from HK and the system changed over to arming.

Note: the plug-in hasn't been very reliable for a while now due to random network errors.  All the other plug-ins work fine, and occasionally report the inability to communicate with their external servers, but they recover and the blips don't last but one request, and the network is up, my ISP says everything is fine, I've done pings for long stretches and don't see any issues.  Subsequently I have a cron task scheduled to kill -9 the xfinity child bridge every 12hrs, and homebridge automatically restarts it.  This has made it better, but not great/perfect.

### To Reproduce:

Random http 500 error.

### Expected behavior:

Hopefully the plug in would restart, or say something indicate it was recovering and able to connect again, and say everything is OK again, etc.

My cron workaround sucks.

### Logs:

```shell
[12/2/2024, 4:34:03 PM] [Xfinity Home] Upstairs: Failed To Fetch Smoke State With Error:  {
  "message": "Request failed with status code 500",
  "name": "Error",
  "stack": "Error: Request failed with status code 500\n    at createError (/var/lib/homebridge/node_modules/homebridge-xfinityhome/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/var/lib/homebridge/node_modules/homebridge-xfinityhome/node_modules/axios/lib/core/settle.js:17:12)\n    at IncomingMessage.handleStreamEnd (/var/lib/homebridge/node_modules/homebridge-xfinityhome/node_modules/axios/lib/adapters/http.js:322:11)\n    at IncomingMessage.emit (node:events:530:35)\n    at endReadableNT (node:internal/streams/readable:1696:12)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)",
  "config": {
    "transitional": {
      "silentJSONParsing": true,
      "forcedJSONParsing": true,
      "clarifyTimeoutError": false
    },
    "transformRequest": [
      null
    ],
    "transformResponse": [
      null
    ],
    "timeout": 8000,
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
    "maxContentLength": -1,
    "maxBodyLength": -1,
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "X-Client-Features": "auth4all,carousel,carousel-devicedeeplink,no-cookie,deeplinkV1",
      "Authorization": "Bearer <redacted>",
      "Xh-Auth": "<redacted>",
      "User-Agent": "axios/0.26.1"
    },
    "baseURL": "https://xhomeapi-lb-prod.codebig2.net/",
    "method": "get",
    "url": "/client/icontrol/device/1112"
  },
  "status": 500
}

I have the unredacted version locally if needed.  I wasn't sure if my credentials were embedded in the log message.
```

### Config:

```shell
{
    "name": "Xfinity Home",
    "refreshToken": "<redacted>",
    "pin": "1597",
    "temperatureSensors": true,
    "logLevel": 3,
    "_bridge": {
        "name": "HB Xfinity Home-Bridge",
        "username": "<redacted>",
        "port": 55260
    },
    "platform": "XfinityHomePlatform"
}
```

### Operating System:

Raspberian

### Node Version:

v20.11.0 

### NPM Version:

10.2.4

### Homebridge Version:

1.8.5

### Plugin Version:

v3.6.0

**Comment by bloomkd46** [comment]
This was the driving reason for me to rewrite this plugin. I had also been using cron to restart my instance on a daily basis because of it suddenly dropping. I did identify that this problem is from Xfinity’s services being down, which is why your ISP couldn’t help. I will work on shortening these errors to be more user friendly, but the plugin should recover once Xfinity is back online, so you shouldn’t need your cron task anymore. If you notice that the plugin is still hanging, let me know. Otherwise, I’m considering this patched with v`3.6.0`

**Comment by rdeutsch3** [comment]
Still working great on 3.6.1-beta2. The fix is transformative.  Well done.
